cylc/flow/task_state.py

```diff
@@ -154,6 +155,75 @@
 }
 
 
+class RunMode:
+    """The possible run modes of a task/workflow."""
+
+    LIVE = 'live'
+    """Task will run normally."""
+
+    SIMULATION = 'simulation'
+    """Task will run in simulation mode."""
+
+    DUMMY = 'dummy'
+    """Task will run in dummy mode."""
+
+    SKIP = 'skip'
+    """Task will run in skip mode."""
+
+    WORKFLOW = 'workflow'
+    """Default to workflow run mode"""
+
+    MODES = {LIVE, SIMULATION, DUMMY, SKIP, WORKFLOW}
+
+    WORKFLOW_MODES = [LIVE, DUMMY, SIMULATION, SKIP]
+    """Workflow mode not sensible mode for workflow.
+
+    n.b. converted to a list to ensure ordering doesn't change in
+    CLI
+    """
+
+    JOB_MODES = {LIVE, DUMMY}
+    """Modes which need to have real jobs submitted."""
+
+    JOBLESS_MODES = {SKIP, SIMULATION}
+    """Modes which completely ignore the standard submission path."""
+
+    @staticmethod
+    def get(options: 'Values') -> str:
+        """Return the run mode from the options."""
+        return getattr(options, 'run_mode', None) or RunMode.LIVE
+
+    @staticmethod
+    def is_lively(mode: str) -> bool:
+        """Task should be treated as live, mode setting mess with scripts only.
+        """
+        return bool(mode in RunMode.JOB_MODES)
+
+    @staticmethod
+    def is_ghostly(mode: str) -> bool:
+        """Task has no reality outside the scheduler and needs no further
+        processing after run_mode.submit_task_job method finishes.
+        """
+        return bool(mode in RunMode.JOBLESS_MODES)
```

Comment:
Unnecessary? And still using the "lively" and "ghostly" terminology which IMO should be avoided